src/context/block_unit.rs

```diff
@@ -256,25 +258,61 @@ impl<'a> BlockContext<'a> {
     }
   }
 
-  pub const fn checkpoint(&self) -> BlockContextCheckpoint {
-    BlockContextCheckpoint {
+  pub fn checkpoint(
+    &self, tile_bo: &TileBlockOffset, chroma_sampling: ChromaSampling,
+  ) -> BlockContextCheckpoint {
+    let bx = tile_bo.0.x & (COEFF_CONTEXT_MAX_WIDTH - MIB_SIZE);
+    let mut bcc = BlockContextCheckpoint {
+      tsbx: tile_bo.0.x,
+      chroma_sampling,
       cdef_coded: self.cdef_coded,
-      above_partition_context: self.above_partition_context,
+      above_partition_context: [0; MIB_SIZE >> 1],
       left_partition_context: self.left_partition_context,
-      above_tx_context: self.above_tx_context,
+      above_tx_context: [0; MIB_SIZE],
       left_tx_context: self.left_tx_context,
-      above_coeff_context: self.above_coeff_context,
+      above_coeff_context: [[0; MIB_SIZE]; MAX_PLANES],
       left_coeff_context: self.left_coeff_context,
+    };
+    bcc.above_partition_context.copy_from_slice(
+      &self.above_partition_context[(bx >> 1)..][..(MIB_SIZE >> 1)],
+    );
+    bcc
+      .above_tx_context
+      .copy_from_slice(&self.above_tx_context[bx..][..MIB_SIZE]);
+    let num_planes =
+      if chroma_sampling == ChromaSampling::Cs400 { 1 } else { 3 };
+    for p in 0..num_planes {
+      let xdec = (p > 0 && chroma_sampling != ChromaSampling::Cs444) as usize;
+      unsafe {
```

Comment:
I'd rather avoid this if it has a measurable impact split the unchecked from the copy_from_slice please.